src/components/location/Location.js

```diff
@@ -0,0 +1,69 @@
+import React from "react"
+import { Link } from "gatsby"
+import { BsArrowLeft } from "react-icons/bs"
+import { MapContainer, TileLayer, Marker, Popup } from "react-leaflet"
+import Layout from "../layout/Layout"
+import {
+  StyledLocationPageContantTop,
+  StyledLocationPageContantTopInfo,
+  StyledLocationPageCt,
+  StyledLocationPageHeader,
+  StyledLocationPageSecoundContant,
+  StyledLocationPageSecoundContantSinlgeItem,
+  StyledLocationPageSecoundContantToggle,
+} from "./LocationStyle"
+import { MdLocationOn } from "react-icons/md"
+import { posts } from "../../data/data"
+
+const Location = ({ post }) => {
+  const selectedPost = post[0]
+  const { location, position } = selectedPost
```

Comment:
why not `const { location, position } = post[0]`? and also if that prop `post` is an array, it means it is plural. so the name should be `posts`